tests/translated_table_model_tester.ts

```diff
@@ -92,31 +92,39 @@ export class TranslatedTableModelTester {
   public testReceiveAdd(): void {
     const model = new ArrayTableModel();
     const translatedTable = new TranslatedTableModel(model);
-    model.addRow([0,0]);
-    model.addRow([1,1]);
+    model.addRow([0]);
+    model.addRow([1]);
     Expect(translatedTable.get(0, 0)).toEqual(0);
     Expect(translatedTable.get(1, 0)).toEqual(1);
     const model2 = new ArrayTableModel();
-    model2.addRow([1, 2]);
-    model2.addRow([3, 4]);
-    model2.addRow([5, 6]);
-    model2.addRow([7, 8]);
+    model2.addRow([1]);
+    model2.addRow([3]);
+    model2.addRow([5]);
+    model2.addRow([7]);
     const translatedTable2 = new TranslatedTableModel(model2);
     translatedTable2.moveRow(2, 0);
-    model2.addRow([0, 0], 1);
+    Expect(translatedTable2.get(0, 0)).toEqual(5);
+    Expect(translatedTable2.get(1, 0)).toEqual(1);
+    Expect(translatedTable2.get(2, 0)).toEqual(3);
+    Expect(translatedTable2.get(3, 0)).toEqual(7);
+    model2.addRow([0], 1);
     Expect(translatedTable2.get(0, 0)).toEqual(5);
     Expect(translatedTable2.get(1, 0)).toEqual(1);
```

Comment:
Lines 99 to 103 create table:
1, 3, 5, 7
Line 105 moves row 2 to 0 resulting in:
5, 1, 3, 7
Line 110 adds row [0] to 1 resulting in:
5, 0, 1, 3, 7

But your test is expecting 5, 1, 0, 3, 7. The test is incorrect.

Reply:
It adds row [0] to 1 in model. [3] is at 1.
 So the [0] should be added above the [3], not above the [1].

Reply:
I realize my comment may have not been clear.
Model: 1, 3, 5, 7
Table: 5, 1, 3, 7
Line 110 adds row [0] to 1 in the array model resulting in:
Model: 1, 0, 3, 5, 7
Table: 5, 1, 0, 3, 7

Reply:
Okay that changes the semantics of how this table was originally intended to work. The original intention was that if a new row is added, its treated as if it was never moved/translated. The way you explained it, if a row is added at index Q, then the translated table adds it at ```this.reverseTranslation[Q]```. That could work out but I'll have to think about it a bit to ensure it doesn't introduce an inconsistency.

I will get back to this sometime tomorrow.

Reply:
Oh wow. That certainly does change things.